include/stl2/detail/semiregular_box.hpp

```diff
@@ -35,61 +37,38 @@ STL2_OPEN_NAMESPACE {
 
 			constexpr semiregular_box(const T& t)
 			noexcept(is_nothrow_copy_constructible<T>::value)
+			requires CopyConstructible<T>
 			: o_{in_place, t} {}
 
+			semiregular_box(const semiregular_box&) requires CopyConstructible<T> = default;
+			semiregular_box(semiregular_box&&) = default;
+
 			template <class... Args>
 			requires Constructible<T, Args...>
 			constexpr semiregular_box(in_place_t, Args&&... args)
 			noexcept(is_nothrow_constructible<T, Args...>::value)
 			: o_{in_place, std::forward<Args>(args)...} {}
 
-			constexpr semiregular_box(const semiregular_box& that)
-			noexcept(is_nothrow_copy_constructible<T>::value)
-			{
-				if (that.o_)
-					o_.emplace(*that.o_);
-			}
-			constexpr semiregular_box(semiregular_box&& that)
-			noexcept(is_nothrow_move_constructible<T>::value)
-			{
-				if (that.o_)
-					o_.emplace(std::move(*that.o_));
-			}
-
 			constexpr semiregular_box& operator=(const semiregular_box& that) &
 			noexcept(is_nothrow_copy_constructible<T>::value)
+			requires CopyConstructible<T>
 			{
+				o_.reset();
 				if (that.o_)
 					o_.emplace(*that.o_);
-				else
-					o_.reset();
-				return *this;
-			}
-			constexpr semiregular_box& operator=(const semiregular_box& that) &
-			noexcept(is_nothrow_copy_constructible<T>::value &&
-				is_nothrow_copy_assignable<T>::value)
-			requires Assignable<T&, const T&>
-			{
-				o_ = that.o_;
 				return *this;
 			}
+			semiregular_box& operator=(const semiregular_box&) & requires Copyable<T> = default;
+
 			constexpr semiregular_box& operator=(semiregular_box&& that) &
 			noexcept(is_nothrow_move_constructible<T>::value)
 			{
+				o_.reset();
 				if (that.o_)
 					o_.emplace(std::move(*that.o_));
-				else
-					o_.reset();
-				return *this;
```

Comment:
Why force a destruct/reconstruct when an assignment might be more efficient?

Reply:
This is only used for types that are not move assignable; they get the `requires Movable<T>` defaulted overload below